### PR TITLE
Tensorboard histogram

### PR DESCRIPTION
**Pull request by cjcchen**
Add tensorboard histogram support
I also add an example about how to use histogram in the example folder.

### CODE REVIEW

**Comment by ryanjulian** [comment]
Looks like there are merge conflicts? You can fix them with
```sh
git checkout integration
git fetch ryanjulian
git reset --hard ryanjulian/integration
git push -f origin integration
git checkout tensorboard_histogram
git rebase integration # resolve conflicts here
git push -f origin tensorboard_integration
```

File: rllab/misc/logger.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 from rllab.misc.tabulate import tabulate
 from rllab.misc.console import mkdir_p, colorize
 from rllab.misc.autoargs import get_all_parameters
+from rllab.misc.tensorboard_summary import Summary
```
Comment (ryanjulian) [inline rllab/misc/logger.py]: please reformat the imports to be PEP8-compliant.

File: rllab/misc/logger.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 from rllab.misc.tabulate import tabulate
 from rllab.misc.console import mkdir_p, colorize
```
Comment (ryanjulian) [inline rllab/misc/logger.py]: one import per line in PEP8

File: examples/example_tensorboard_logger.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import numpy as np
+from rllab.misc import logger
```
Comment (ryanjulian) [inline examples/example_tensorboard_logger.py]: please use PEP8-compliant import ordering

File: rllab/misc/tensorboard_summary.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
+from rllab.misc.console import mkdir_p
+import tensorflow as tf
```
Comment (ryanjulian) [inline rllab/misc/tensorboard_summary.py]: use PEP8 import ordering/grouping

**Review by ryanjulian — COMMENTED** [review]
```sh
(rllab3-tbh) rjulian@nyquist:~/code/test_tbh/rllab$ python examples/example_tensorboard_logger.py 
tensorboard data will be logged into: ./test
Traceback (most recent call last):
  File "examples/example_tensorboard_logger.py", line 9, in <module>
    logger.record_histogram("gass", val)
AttributeError: module 'rllab.misc.logger' has no attribute 'record_histogram'
(rllab3-tbh) rjulian@nyquist:~/code/test_tbh/rllab$ 
```

**Comment by ryanjulian** [comment]
```sh
2018-04-06 10:05:10.924477 PDT | app  98
2018-04-06 10:05:10.924712 PDT | ---  --
2018-04-06 10:05:10.940285 PDT | ---  --
2018-04-06 10:05:10.940501 PDT | app  99
2018-04-06 10:05:10.940637 PDT | ---  --
Exception ignored in: <bound method BaseSession.__del__ of <tensorflow.python.client.sess
ion.Session object at 0x7fae3404d860>>
Traceback (most recent call last):
  File "/home/rjulian/miniconda2/envs/rllab3-tbh1/lib/python3.5/site-packages/tensorflow/
python/client/session.py", line 712, in __del__
  File "/home/rjulian/miniconda2/envs/rllab3-tbh1/lib/python3.5/site-packages/tensorflow/
python/framework/c_api_util.py", line 31, in __init__
TypeError: 'NoneType' object is not callable
```

**Comment by cjcchen** [comment]
This error seems a bug in tensorflow. 
> https://github.com/tensorflow/tensorflow/issues/8652

But it just occurs while calling the GC.

**Comment by ryanjulian** [comment]
Okay good catch.

**Comment by eric-heiden** [comment]
It would be nice to support histograms for distributions as shown here: https://www.tensorflow.org/programmers_guide/tensorboard_histograms.

```python
# Make a normal distribution, with a shifting mean
mean_moving_normal = tf.random_normal(shape=[1000], mean=(5*k), stddev=1)
# Record that distribution into a histogram summary
tf.summary.histogram("normal/moving_mean", mean_moving_normal)
```
Then concatenating different distributions:
```Python
all_distributions = [mean_moving_normal, variance_shrinking_normal,
                     gamma, poisson, uniform]
all_combined = tf.concat(all_distributions, 0)
tf.summary.histogram("all_combined", all_combined)
```
I would add a `record_distributions("name", [dist1, dist2, ...])` function that visualizes these distributions where `dist1` could be a `tf.random_normal(shape=[1000], mean=(5*k), stddev=1)` etc.
That comes closer to our use case of histograms where we have to visualize distributions (not just numpy arrays).

**Comment by ryanjulian** [comment]
This seems to have a bunch of DDPG commits which don't belong.

**Comment by cjcchen** [comment]
I made a new PR since I made something wrong  during the rebase. I could not find the way to reset the rebase.

**Comment by ryanjulian** [comment]
Which PR should I look at? This one or #58?

**Pull request closed without merging**